### ISSUE TITLE
Means to manage the created fiddles

### ISSUE THREAD

**Issue by soumyahameed** [tony1223/zkfiddle#181]
Hello there... Being an extensive user of ZK, one of the main issues which I face while using ZK Fiddle is that there is no way to manage the previously created fiddles by:

I. Updating an existing version of a fiddle without resulting in a new version.
2. Deleting those fiddles which are no longer in use.

I find that the lack of the above features results in many fiddle versions which varies only slightly from each other. 

So basically, what I am suggesting is to have the following options:
1. A 'Replace' button, along side the 'Update' button on the fiddle page, for replacing the existing contents in a particular fiddle version with the new contents and without creating a new version.
2. The fiddle list in the user's home page can have select boxes for each fiddle and a 'Delete' button on the top to delete the selected fiddles.

**Comment by tony1223** [comment]
Actually it's designed in this way.

Just like gist , you could not replace any existing version,
and you don't have to care if there's any duplicated fiddle. (Or so-called "fork".)

The benefits is we could always make sure the fiddle is the version when we submit,
and we don't need to worry about the permission part when editing/deleting. 

If a fiddle is no longer in use , it's ok . Just left it there. That won't hurt the system.

What fiddle need is a good way to organize the fiddles. 
(In the beginning design , it should be organized in other service...like ZK demo.)

It's not really designed to be a content provider , it's just a paste-bin.
Upload a code and run it,that's how it work. :)

Related to this issue, I think there might be a missing feature for a link to fetch last version , like how gist does.

**Comment by soumyahameed** [comment]
ZK Fiddle is a wonderful website, especially because of the convenience it gives to test a ZK functionality before adopting it in a project. I understand its a paste-bin, but since one of its main uses is to run a ZK functionality immediately, it always results in multiple versions of the functionality as you will continuously fine tune the code to get the desired output. And the consequence is that when you come back later to share this tested functionality with one of your colleagues or ZK users out there, you are confused as to which one was the fine tuned version. 

So I keep wondering wouldn't a better way to organize the fiddles help the users? I mean if providing editing/deleting features involves taking care of the permissions part (though I feel it wouldn't be that difficult given that something similar is already done for retrieving the fiddles created by a particular user for display in the user's home page), at least a feature to insert comments for the fiddle or as you suggested, a link to fetch the previous version would definitely be nice to have. This would help the user creating the fiddle as well as the other users who are looking up for a particular fiddle. 

Let me know your thoughts.

**Comment by tony1223** [comment]
It's a lot of side impact if we did edit/delete functionality, like caching/sandbox ...etc ,I don't think that's a good way to go that if we expect ZK fiddle will keep growing. So I think it's not top priority now, at least not now.

For the confused ZK version, yes , that's a real issue. It's getting more difficult after ZK 6, that's a major issue in my mind so far. In the beginning ideas, there should be a properties (or tag) for available(suggested) ZK versions of the fiddle, when you set up the properties. The version of the sandbox will be filtered out with the given properties(or tag).

Yep , that's just a idea. Sadly , I still haven't time to implement it.I haven't commit any code recently and actually I don't have the server access right after I quit ZK six months ago, so there's still a lot of pending features. (Now the official mantianer is Ian as far I know.  https://github.com/zanyking/zkfiddle ) 

That's a good idea to write some code for previous link,and the comment, I will try to write some code and talk to Ian.

**Comment by soumyahameed** [comment]
Thanks. Meanwhile, I will download the code and give it a try myself.
